RFC-0005-dyllib-cpp.md

```diff
@@ -0,0 +1,87 @@
+## Creating a Dynamically Linked Functions Library in CPP
+
+Proposers
+
+* Soumya Duriseti
+* Tim Meehan
+
+## [Related Issues]
+
+https://github.com/facebookincubator/velox/pull/1005
+
+## Summary
+
+This proposed change expands the dynamic function loading ability to cpp user defined functions (UDFs). The Prestissimo worker is to access said code. The dynamic functions are to be loaded upon running an instance of the presto server. In the presto server instance, it will search for any .so or .dylib files and load them using this library.
+## Background
+
+Currently, on Presto, any Java UDFs can be loaded dynamically. This is an important feature for any client who wants their Presto build to be lightweight and to protect any non-public code or information from having to be in the open source space. Creating UDFs is an important functionality Presto has and its competitors offer as well. Being able to load CPP functions dynamically will extend the offering in Prestissimo.
+
+### [Optional] Goals
+Should be able to register all functions on runtime as theyre specified in the plugin directory. 
+### [Optional] Non-goals
+Security concerns: There are some security concerns associated with using the built in plugins. Using the dlopen library, we run the risk of opening unsecure unknown shared objects especially given the lack of any form of validation. On CPP, we share the same limitations as the functionality in Java with a noted exception of Java Presto running in a VM environmnet while the CPP version will be run locally. We will not be addressing these security concerns through code at this time.
+
+## Proposed Implementation
+The user can register their functions dynamically by calling loadDynamicLibraryFunctions() with the path to their shared library (files ending in .so in linux or .dylib in MacOS). At the time of running an instance of the PresterServer, if any shared library files exist in plugin directory, they get loaded on start up. Alternatively, the user can call loadDynamicLibraryFunctions() elsewhere and specify the exact location of these files and load them upon execution of this code.
```

Comment:
Does the user call `loadDynamicLibraryFunctions`?

Reply:
the call to loadDynamicLibraryFunctions() would happen in presto server by design so user doesnt need to do anythign there excecpt put the plugin dir path in their config.properties. however theyre free to call loadDynamicLibraryFunctions too if they want to